docs/hybrid_applications.md

```diff
@@ -0,0 +1,71 @@
+# Hybrid applications
+
+## Overview
+
+You can use Datadog on the same application both from the React Native and the native sides.
+
+RUM Events will be reported as coming from the same application and the same source in Datadog RUM.
+
+## Constraints
+
+A few constraints to keep in mind:
+
+1. You cannot initialize more than 1 instance of the native `DatadogSDK` as it uses a singleton pattern on the native side. That means that once you initialize the native SDK on the native side or on the RN side (by calling `DdSdkReactNative.initialize`) it is configured for both.
+2. If you try to report Datadog RUM events or logs before the initialization on iOS, future RUM events and logs won't be sent.
+3. For **errors, resources and interactions tracking**, SDKs can work in 2 ways: through auto-instrumentation (some React classes and methods are modified to automate this) or through manual instrumentation - for instance if you want to report something you consider an error but that is not going to crash the app. Auto-instrumentation for JS errors, resources and interactions can only be started from javascript code.
```

Comment:
I'm not sure if this should be included in the hybrid apps documentation as there really isn't anything particular to hybrid apps here. It's just how the SDK works for all RN apps.

Reply:
Indeed it is the way the SDK works for every app, but that is not mentioned anywhere else in the documentation and I think it brings value to mention it here so that developers understand the difference between auto-instrumentation and manual instrumentation.

Reply:
However I think it makes more sense to mention this first to add more details as per your first comment